### DISCUSSION TITLE
How to Back Up Windows 10 Start Menu Layout

### DISCUSSION THREAD

**Discussion by ghost** [valinet/ExplorerPatcher#2620] [General]
One of the more time consuming routines after "refreshing" the OS and reinstalling ExplorerPatcher is getting the Windows 10 start menu back to the way I like it.

I'm looking at an instructional page at [Major Geeks](https://www.majorgeeks.com/content/page/how_to_backup_and_restore_the_start_menu_in_windows_10.html) on how to back everything up.

The page states that two backups are required; one for the registry tree under `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount`, and another for the DefaultLayouts.xml located in `C:\Users\USERNAME\AppData\Local\Microsoft\Windows\Shell`

~[Another page](https://www.majorgeeks.com/content/page/backup_restore_pinned_items_start_menu.html) mentions backing up the start.bin file located in `%LocalAppData%\Packages\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\LocalState`~ Start.bin file is for backing up Windows 11 start menu items.

I just wanted to check here to see if anyone can confirm that this will fully back up everything needed for the pinned tiles of the Windows 10 start menu to show up exactly as I have set them.

**Comment by pyrates999** [comment]
Look here for how to export and import settings for EP:

![image](https://github.com/valinet/ExplorerPatcher/assets/106135298/72cc295c-8f82-4856-b05c-938d41e68d2c)

**Reply by ghost** [reply]
Thanks for the reply. I've used the export/import a few times but it doesn't seem to affect the placement of pinned tiles on the W10 start menu.

I just want to make sure I get EVERYTHING backed up for the next system wipe.

**Reply by pyrates999** [reply]
EP doesn't keep the settings for the pinned tiles on the windows 10 start menu.  That's all stored by Windows.

**Reply by ghost** [reply]
So on a W11 machine, does the EP start menu (W10 style) work exactly the way it would in W10?

**Reply by pyrates999** [reply]
It should.  There's options to change some of the settings in the EP properties application when you do change it to windows 10 start menu since.

**Reply by ghost** [reply]
Part of the problem is that I'm additionally using [TileIconifier](https://github.com/Jonno12345/TileIconifier) to mod the tiles themselves, so I have to restore ExplorerPatcher and TileIconifier settings after a system reset, and then spend potentially a couple of hours messing around with pinning and placement, especially with [EP's quirks](https://github.com/valinet/ExplorerPatcher/discussions/1679).

So if I can automate that process, so much the better.

On further research, it looks like the exact registry key the Windows 10 tile placements are kept in might be `Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount\$de${YOUR_GUID_GOES_HERE}$start.tilegrid$windows.data.curatedtilecollection.tilecollection\Current`

Interestingly, because of how TileIconifer works, I do not need to first reinstall the applications to ensure their pinned tile will reappear, giving me the ability to do that later at my leisure.

**Reply by pyrates999** [reply]
Let us know if it works.

**Reply by pacopepepipo** [reply]
did you solve this?

I am on the very exact same position.

Windows 10 but I am going to migrate to windows 11, I wanted to save my windows 10 start menu layout before migrating, so I can restore it later without spending hours adding the 150 icons from my start menu

I also had some of them with tileiconifier, that wouldn't be much problem because they are like 10, but migrating the W10 icons to ExplorerPatcher would be nice

**Reply by ghost** [reply]
> did you solve this?
> 
> I am on the very exact same position.
> 
> Windows 10 but I am going to migrate to windows 11, I wanted to save my windows 10 start menu layout before migrating, so I can restore it later without spending hours adding the 150 icons from my start menu
> 
> I also had some of them with tileiconifier, that wouldn't be much problem because they are like 10, but migrating the W10 icons to ExplorerPatcher would be nice

I made a PowerShell script that will backup the files that are needed for the Start Menu tile layout:

```# Define the paths
$backupDir = "path\to\backup\folder"
$regKeyPath = "HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount"
$xmlFilePath = "$env:LocalAppData\Microsoft\Windows\Shell\DefaultLayouts.xml"

# Create the backup directory if it doesn't exist
if (-not (Test-Path -Path $backupDir)) {
    New-Item -ItemType Directory -Path $backupDir
}

# Backup the registry key
$regBackupFile = Join-Path $backupDir "DefaultAccount.reg"
reg export "$regKeyPath" "$regBackupFile" /y

# Backup the XML file
$xmlBackupFile = Join-Path $backupDir "DefaultLayouts.xml"
Copy-Item -Path $xmlFilePath -Destination $xmlBackupFile -Force

Write-Host "Backup completed successfully."
```

Specify the backup folder, save the file as a .ps1, and then run it whenever you need to create backups.

**Reply by pacopepepipo** [reply]
gotcha, and then is there any way to use that backup on W11 or explorer patcher so the tiles are automatically ordered from that backup?

**Reply by ghost** [reply]
The registry backup you can double-click to add it to registry, and then put the XML file in the `C:\Users\user\AppData\Local\Microsoft\Windows\Shell` folder.

When you set ExplorerPatcher to use the W10 style start menu, EP will set your Windows OS to use those restored registry entries and the restored XML. W10 and W11 have completely different systems for running the taskbar, start menu, etc. You can have settings for both on your system, but EP controls which one will be displayed.
